src/file.ts

```diff
@@ -1896,10 +1893,7 @@ class File extends ServiceObject<File> {
       let failed = crc32c || md5;
 
       if (crc32c && metadata.crc32c) {
-        // We must remove the first four bytes from the returned checksum.
-        // http://stackoverflow.com/questions/25096737/
-        //   base64-encoding-of-crc32c-long-value
-        failed = !validateStream.test('crc32c', metadata.crc32c.substr(4));
+        failed = !validateStream.test('crc32c', metadata.crc32c);
```

Comment:
Any reason you know of that this was previously removing the first 4 bytes? I think you mentioned it was only checking those bytes?

Reply:
Yea… unfortunately we weren’t checking the complete hash before - just the last byte

Reply:
In short, invalid implementation - see the link from the removed comment:
- http://stackoverflow.com/questions/25096737/base64-encoding-of-crc32c-long-value

This would result in inaccurate validation. E.g.:
Before: https://github.com/googleapis/nodejs-storage/blob/main/test/file.ts#L2373
After: https://github.com/googleapis/nodejs-storage/blob/06e92ac57694ba259379c13383a9f5bd7ed56021/test/file.ts#L2194

The perf for validation is the same - the `update` method is the sensitive portion for performance